### PR TITLE
spec: adhere to "one sentence per line" formatting

### LINKED ISSUE #4: spec: format according to opencontainers guidelines
After import, I'll take an initial pass over the spec to make sure we use the "one sentence per line" guideline. There are definitely other formatting requirements but this one will be time consuming.

### PR DESCRIPTION
**Pull request by jzelinskie**
Fixes #4.

Thank god for sed and vim macros.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -273,141 +236,123 @@ errors will be returned in the following format:
...
+The _algorithm_ identifies the methodology used to calculate the digest.
+The _hex_ portion is the hex-encoded result of the hash.
 
 We define a _digest_ string to match the following grammar:
+
 ```
 digest      := algorithm ":" hex
-algorithm   := /[A-Fa-f0-9_+.-]+/
+algorithm   := /[A-Fa-f0-9_+.
+-]+/
```
Comment (dmcgowan) [inline spec.md]: unintended new line

File: spec.md [hunk 2]
```diff
@@ -456,59 +396,51 @@ The image manifest can be checked for existence with the following url:
...
 
-All layer uploads use two steps to manage the upload process. The first step
-starts the upload in the registry service, returning a url to carry out the
-second step. The second step uses the upload url to transfer the actual data.
-Uploads are started with a POST request which returns a url that can be used
-to push data and check upload status.
+All layer uploads use two steps to manage the upload process.
+The first step starts the upload in the registry service, returning a url to carry out the second step.
+The second step uses the upload url to transfer the actual data.
+Uploads are started with a POST request which returns a url that can be usedto push data and check upload status.
```
Comment (dmcgowan) [inline spec.md]: `usedto` -> `used to`

File: spec.md [hunk 3]
```diff
@@ -933,48 +807,40 @@ Link: <<url>?n=<n from the request>&last=<last repository in response>>; rel="ne
 {
   "repositories": [
     <name>,
-    ...
+    .
```
Comment (dmcgowan) [inline spec.md]: unintended new lines

File: spec.md [hunk 4]
```diff
@@ -3623,44 +3145,38 @@ Range: 0-0
...
+| `DIGEST_INVALID` | provided digest did not match uploaded content | When a blob is uploaded, the registry will check that the content matches the digest provided by the client.The error may include a detail structure with the key "digest", including the invalid digest string.This error may also be returned when a manifest includes an invalid layer digest. |
+| `NAME_INVALID`   | invalid repository name                        | Invalid repository name encountered either during manifest validation or any API operation.                                                                                                                                                                                                       |
 
 ###### On Failure: Authentication Required
 
 ```
 401 Unauthorized
-WWW-Authenticate: <scheme> realm="<realm>", ..."
+WWW-Authenticate: <scheme> realm="<realm>", ...
+"
```
Comment (dmcgowan) [inline spec.md]: unintended new line

File: spec.md [hunk 5]
```diff
@@ -3672,28 +3188,24 @@ Content-Type: application/json; charset=utf-8
             "detail": ...
         },
         ...
-    ]
+  ]
```
Comment (dmcgowan) [inline spec.md]: I don't think this fomatting change is correct

File: spec.md [hunk 6]
```diff
@@ -4416,18 +3845,14 @@ Content-Type: application/json; charset=utf-8
 }
 ```
 
-The upload is unknown to the registry. The upload must be restarted.
-
-
+The upload is unknown to the registry.The upload must be restarted.
```
Comment (dmcgowan) [inline spec.md]: Should have a new line here

**Comment by dmcgowan** [comment]
Thanks for doing this, just a couple formatting corrections!

**Comment by dmcgowan** [comment]
Update looks good, can you please sign the commit

**Comment by jzelinskie** [comment]
Oh! Did you mean `git -s` because I did `git -S`, haha! I figured a GPG Signature is better, but sure thing. I can do both.

**Comment by dmcgowan** [comment]
LGTM

**Comment by vbatts** [comment]
LGTM

**Pull request merged**